com.kms.cura.dal/src/com/kms/cura/dal/database/AppointmentDatabaseHelper.java

```diff
@@ -0,0 +1,103 @@
+package com.kms.cura.dal.database;
+
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Set;
+
+import com.kms.cura.dal.mapping.AppointmentColumn;
+import com.kms.cura.dal.mapping.DoctorColumn;
+import com.kms.cura.dal.mapping.PatientColumn;
+import com.kms.cura.entity.AppointmentEntity;
+import com.kms.cura.entity.Entity;
+import com.kms.cura.entity.user.DoctorUserEntity;
+import com.kms.cura.entity.user.PatientUserEntity;
+
+public class AppointmentDatabaseHelper extends DatabaseHelper {
+
+	public AppointmentDatabaseHelper() throws ClassNotFoundException, SQLException {
+		super();
+	}
+
+	@Override
+	protected Entity getEntityFromResultSet(ResultSet resultSet) throws SQLException, ClassNotFoundException {
+		return null;
+	}
+
+	protected AppointmentEntity geAppointmentEntityFromResultSet(ResultSet resultSet, PatientUserEntity patientUserEntity, DoctorUserEntity doctorUserEntity)
```

Comment:
This method is way too long and have duplicated codes. You should create variable to hold patient and doctor, then only call constructor once